admin/services/status/status.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2019 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package update implements update API.
+package status
+
+import (
+	"context"
+
+	"github.com/percona/pmm/api/updatepb"
+)
+
+type Server struct {
+	updatepb.UnimplementedStatusServer
+}
+
+// New returns new instance of Server.
+func New() *Server {
+	return &Server{}
+}
+
+// Ready returns no error if the daemon is ready.
+func (s *Server) Ready(ctx context.Context, req *updatepb.ReadyRequest) (*updatepb.ReadyResponse, error) {
```

Comment:
should we check if we have access to docker and if we can find the docker container of PMM since it's a `Ready` function, not just `Available`?

Reply:
Originally I expected to perform the actions you suggested in `Ready`.
Later I realized it may be better to have this in the `update` flow instead. The advantage is we can provide more information what's wrong through the logs.

If we were to return we're not "ready" because the Docker socket is not available, the only way to check for the user what's wrong is to go through the logs of the container.

Looking for feedback here though.

Reply:
I suggest we stick with `Ready`.
I expect there's going to be more logic over time. If there was only `Available` and more logic is added, we need to introduce `Ready` and will need to change API calls. This approach seems easier to maintain in the future even though I agree, at this point it's technically more about availability than readiness.

Reply:
With any new logic coming here we can use your previous statement that it's better to check logs during the start of the upgrade.
I'm for making it `Ready`, but include check for docker, otherwise it's not ready.

Reply:
Ok. Renamed to `Available`